src/challenge.js

```diff
@@ -1,10 +1,29 @@
 const { Deque } = require ('./deque.js');
 
 function palindromeChecker(text) {
+  if (!text) {
+    return false
+  }
 
-  // your code
+  const re = /[^A-Za-z0-9]/g;
+  text = text.toLowerCase().replace(re, '')
 
-  return;
+  let Deq = new Deque()
+  for (var i = 0; i < text.length; i++)
+  {
+    Deq.addFront(text.charAt(i))
```

Comment:
Usa `addBack()` es tiene un algoritmo más rápido O(1). Como `addFront` en este caso es del orden `O(n)`, combinando con el `for` se vuelve en total del orden O(n^2).